### PR TITLE
Initial clean up of database and standardizing names

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:

This PR converges the naming conventions between the initial database implementation and the recently merged yaml files. This also adds some basic tests for the content of the yaml files to ensure consistency.

Currently, there are some failures due to issues with the yaml file, mostly cases of missing `recovery_vol`.

- [x] injection_well_use
- [x] oil_and_ga
- [x] tuscarora_md
- [x] electrodialysis_reversal: pH max and min showing up in removal fractions with negative values. Also has both "nitrate' and "nitrate_as_nitrogen" entries.
- [x] injection_well_treat
- [x] disinfection
- [x] brine_concentrator_crystallizer
- [x] pumping_station

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by kurbansitterley** [comment]
Here is a list of basic_units that are not used in any current WT3 study and do not need to be moved imo. Note that these are **basic_units** (from the basic_unit.csv) and in some cases there are separate modules from WT3 that we _would_ want to keep (big example is chlorination or deep_well_injection). But I believe all the failures are from this list, so just don't include them. 

- anti_oxidant_addition
- brine_concentrator_crystallizer
- brine_ro
- cas_denitrif
- cas_nitrif
- chlorination
- coagulant_addition
- coal_plant_cooling_tower
- crystallizer_waterreuse
- decarbonator_a
- decarbonator_b
- decarbonator_waterreuse
- deep_bed_filtration
- deep_well
- deep_well_injection
- direct_injection
- disinfection
- electro_deionization
- electrodialysis_reversal
- engineered_buffer_with_chloramine
- engineered_buffer_with_chlorine
- evaporation_pond_waterreuse
- fc_crystallizer
- gac_feed_pump
- injection_well_treat
- injection_well_use
- irwin_brackish
- irwin_brine
- irwin_brine_management
- mbr_denitrification
- microfiltration
- oil_and_gas
- onsite_storage
- onsite_storage_old
- pumping_station
- soil_aquifer_treatment
- static_mixer
- sulfuric_acid_addition
- surface_augmentation
- tuscarora_md

**Review by kurbansitterley — DISMISSED** [review]
LGTM (but my experience writing tests is limited)

File: watertap/core/tests/test_wt_database.py [hunk 1]
```diff
@@ -54,6 +54,14 @@ class TestDatabase():
     def db(self):
         return Database()
 
+    @pytest.mark.unit
+    def test_component_list(self, db):
+        # First, check that _component_list exists and is None
+        assert db._component_list is None
+
+        # Then check that component list retreive expected dict
```
Comment (adam-a-a) [inline watertap/core/tests/test_wt_database.py]:
```suggestion
        # Then check that component list retrieved is a dict
```

File: watertap/data/techno_economic/aeration_basin.yaml [hunk 2]
```diff
@@ -1,60 +1,60 @@
...
     viruses_enteric:
       value: 0.99
-      unit: dimensionless
+      units: dimensionless
       constituent_longform: Viruses (enteric)
     total_coliforms_fecal_ecoli:
       value: 0.998798
-      unit: dimensionless
+      units: dimensionless
       constituent_longform: "Total Coliforms (including fecal coliform and\uFFFD E.\
```
Comment (adam-a-a) [inline watertap/data/techno_economic/aeration_basin.yaml:59]: Not sure what was in place of this \uFFFD.

File: watertap/data/techno_economic/cas_nitrif.yaml [hunk 3]
```diff
@@ -1,56 +1,56 @@
...
       function f(flow_vol_in): cap_basis_par*(flow_vol_in/flow_basis_par)^cap_exp
-      unit: $
+      units: $
   recovery_vol:
     value: 0.9999
-    unit: dimensionless
-    reference: 'EPA model: https://www.epa.gov/sdwa/drinking-water-treatment-technology-unit-cost-models'
-  default_removal_mass_solute:
+    units: dimensionless
+    reference: 'EPA model: https://www.epa.gov/sdwa/drinking-water-treatment-technology-units-cost-models'
```
Comment (adam-a-a) [inline watertap/data/techno_economic/cas_nitrif.yaml]:
```suggestion
    reference: 'EPA model: https://www.epa.gov/sdwa/drinking-water-treatment-technology-unit-cost-models'
```

File: watertap/data/techno_economic/gac_feed_pump.yaml [hunk 4]
```diff
@@ -1,30 +1,30 @@
...
       function f(flow_vol_in): cap_basis_par*(flow_vol_in/flow_basis_par)^cap_exp
-      unit: $
+      units: $
   recovery_vol:
     value: 0.9999
-    unit: dimensionless
-    reference: 'EPA model: https://www.epa.gov/sdwa/drinking-water-treatment-technology-unit-cost-models'
-  default_removal_mass_solute:
+    units: dimensionless
+    reference: 'EPA model: https://www.epa.gov/sdwa/drinking-water-treatment-technology-units-cost-models'
```
Comment (adam-a-a) [inline watertap/data/techno_economic/gac_feed_pump.yaml]:
```suggestion
    reference: 'EPA model: https://www.epa.gov/sdwa/drinking-water-treatment-technology-unit-cost-models'
```

File: watertap/data/techno_economic/mining.yaml [hunk 5]
```diff
@@ -1,30 +1,30 @@
...
       function f(flow_vol_in): cap_basis_par*(flow_vol_in/flow_basis_par)^cap_exp
-      unit: $
+      units: $
   recovery_vol:
     value: 0.9999
-    unit: dimensionless
-    reference: 'EPA model: https://www.epa.gov/sdwa/drinking-water-treatment-technology-unit-cost-models'
-  default_removal_mass_solute:
+    units: dimensionless
+    reference: 'EPA model: https://www.epa.gov/sdwa/drinking-water-treatment-technology-units-cost-models'
```
Comment (adam-a-a) [inline watertap/data/techno_economic/mining.yaml]:
```suggestion
    reference: 'EPA model: https://www.epa.gov/sdwa/drinking-water-treatment-technology-unit-cost-models'
```

File: watertap/unit_models/zero_order/nanofiltration_zo.py [hunk 6]
```diff
@@ -39,13 +39,13 @@ def build(self):
         self.electricity = Var(self.flowsheet().time,
                                units=pyunits.kW,
                                doc="Electricity consumption of unit")
-        self.electricity_intensity = Var(
+        self.energy_electric_flow_vol_inlet = Var(
             units=pyunits.kWh/pyunits.m**3,
             doc="Electricity intensity of unit")
```
Comment (adam-a-a) [inline watertap/unit_models/zero_order/nanofiltration_zo.py]:
```suggestion
            doc="Electricity intensity with respect to inlet flowrate of unit")
```

**Review by adam-a-a — APPROVED** [review]
LGTM- just a very minor comment about "\uFFFD" in the aeration_basin.yaml under consitutent_longform for total_coliforms_fecal_ecoli. Some character wasn't recognized. We should replace this at some point.

**Review by kurbansitterley — APPROVED** [review]
LGTM

**Pull request merged**